bpf/flows.c

```diff
@@ -78,6 +78,20 @@ static inline void update_dns(additional_metrics *extra_metrics, pkt_info *pkt,
     }
 }
 
+static inline void add_observed_intf(additional_metrics *value, u32 if_index, u8 direction) {
+    if (value->nb_observed_intf < MAX_OBSERVED_INTERFACES) {
+        for (u8 i = 0; i < value->nb_observed_intf; i++) {
+            if (value->observed_intf[i].if_index == if_index &&
```

Comment:
we should check `if_index` make sure its not 0

Reply:
hmm do you think I should also ignore setting `if_index_first_seen` to `skb->ifindex` if it's 0 ?
(in 
```
        flow_metrics new_flow = {
            .if_index_first_seen = skb->ifindex,
            .direction_first_seen = direction,
            .packets = 1,
            .bytes = len,
// ...
```
)

Reply:
(idk in what cases it would be 0 and if it's something to handle in a special way)

Reply:
added `else if (skb->ifindex != 0)` earlier in the code path

Reply:
there are certian hooks that don't populate if_index u can see examples of that in the current code where we skip populating flow if

Reply:
ok but here it's the tc hook ; maybe that's something not relevant now that other hooks use a different map?  anyway it won't hurt to keep these checks, I guess...